messaging.rst

```diff
@@ -104,38 +104,113 @@ Metadata
 
 .. _metadata:
 
-Message metadata, at the moment only routing information is supported.
+Message metadata that can be used to provide additional information to a transfer.
+A route with additional ``routes`` metadata related to the transport discovery has to be provided
+from the initiator of the payment.
+
+The ``secret`` of the corresponding LockedTransfer ("LT") can optionally already be sent along by including it to
+the LT's ``metadata``. In order to not compromise the security-properties of the LT, the secret has 
+to be encrypted in a specific way, so that only the target of the LT can decrypt it.
+If the encrypted secret is included in the ``Metadata``, the transfer can't be cancelled by the initiator by not 
+revealing the secret via the additionally required :ref:`RevealSecret <reveal-secret-message>` message.
+
+..
+
+The secret should be encrypted with the public key of the LT's target, using the ECIES encryption scheme (look e.g. `here <https://ecies.org/>`__ for reference implementations).
+The encrypted data has to be encoded as an object containing secret and additional metadata needed for validation::
+
+        secret = encrypt_ecies(target_publickey, { "secret": <bytes32>, "amount": <int32>, "payment_identifier": <int32> })
+
+Note the encoding of the encrypted object's field is not part of this specification.
+
+When the target receives a transfer with an encrypted secret, it may try to decrypt it::
+
+        { "secret": <bytes32>, "amount": <int32>, "payment_identifier": <int32> } = encrypt_ecies(target_privatekey, secret)
+
+
+The target should only accept the decrypted secret, if the decrypted objects fields comply to the following properties:
+
+- ``secret`` must hash to the LT's ``lock.secrethash``
+- ``amount`` must be less than or equal to the received LT's ``lock.amount``
+- ``payment_identifier`` must be equal to LT's ``payment_identifier``
+
+If the validation passes, the target uses "secret" as this transfer's secret and skips sending a :ref:`SecretRequest <secret-request-message>` to the initator.
+Therefore the target can immediately start the unlock phase by sending a corresponding :ref:`RevealSecret <reveal-secret-message>` to the last mediator in the transfer's route.
+
```

Comment:
```suggestion
Otherwise, the usual ``SecretRequest`` protocol should be performed as a fallback; even in unrecoverable cases like not enough `amount` (meaning mediators took more than initiator expected/were willing to pay), sending the received-by-target payment info back to initiator in ``SecretRequest``'s payload is useful to inform initiator if and why this transfer failed, and allow them to notify user.

```